pkg/otlp/model/translator/consumer.go

```diff
@@ -16,20 +16,51 @@ package translator
 
 import (
 	"context"
+	"encoding"
+	"fmt"
 
 	"github.com/DataDog/datadog-agent/pkg/quantile"
 )
 
 // MetricDataType is a timeseries-style metric type.
 type MetricDataType int
 
+var _ encoding.TextUnmarshaler = (*MetricDataType)(nil)
+var _ encoding.TextMarshaler = (MetricDataType)(Gauge)
+
 const (
 	// Gauge is the Datadog Gauge metric type.
 	Gauge MetricDataType = iota
 	// Count is the Datadog Count metric type.
 	Count
 )
 
+// UnmarshalText unmarshals a type into its lowercase type name.
+func (t *MetricDataType) UnmarshalText(text []byte) error {
+	switch string(text) {
+	case "gauge":
+		*t = Gauge
+		return nil
+	case "count":
+		*t = Count
+		return nil
+	}
+
```

Comment:
I would remove this empty line, if we are to follow our OTel coding conventions.

Reply:
Ok,
```suggestion
```